misc-utils/getopt.c

```diff
@@ -275,6 +275,18 @@ static void add_longopt(struct getopt_control *ctl, const char *name, int has_ar
 }
 
 
+static void add_short_options(struct getopt_control *ctl, char *options)
+{
+	if (*options != '+' && getenv("POSIXLY_CORRECT")) {
+		ctl->optstr = xmalloc(strlen(options) + 2);
+		ctl->optstr[0] = '+';
+		strcpy(ctl->optstr + 1, options);
+	} else {
+		ctl->optstr = xstrdup(options);
+	}
+}
+
```

Comment:
Maybe use strappend() from include/strutils.h and keep free() in the function too.

```C
static void add_short_options(struct getopt_control *ctl, char *options)
{
  free(ctl->optstr);

  if (*options != '+' && getenv("POSIXLY_CORRECT")) {
     ctr->optstr = strappend("+", options);
     if (!ctr->optstr)
        err_oom();
  } else
     ctl->optstr = xstrdup(options);
}
```

Reply:
I'm not sure if it make sense or not. We may need to `err_oom` on `xstrdup`, too?

Reply:
Yes, it's possible (and it makes the code more redable), but in this the case old good strdup() is enough as xstrdup() internally checks for NULL and never returns it (all include/xalloc.h functions always succes or exit()).

Reply:
Ah, yes. I misread that part in `include/xalloc.h`. Anyway, I went with that route, since the code is more readable. If you don't like it, I can revert to your suggestion.